presto-main/src/main/java/com/facebook/presto/memory/QueryContext.java

```diff
@@ -388,6 +402,11 @@ public synchronized void setMemoryLimits(DataSize queryMaxTaskMemory, DataSize q
         memoryLimitsInitialized = true;
     }
 
+    public synchronized void setVerboseExceededMemoryLimitErrorsEnabled(boolean verboseExceededMemoryLimitErrorsEnabled)
+    {
+        this.verboseExceededMemoryLimitErrorsEnabled = this.verboseExceededMemoryLimitErrorsEnabled || verboseExceededMemoryLimitErrorsEnabled;
```

Comment:
Can you please explain why this setter is using an OR? The value of `verboseExceededMemoryLimitErrorsEnabled` is coming from a session property which should be constant for the entire duration of the query. Right?

Reply:
Yeah, in practice all tasks of the same query should have the same session properties. But unfortunately it is not represented in the interface. Let me remove the `OR` since it makes the code even more confusing.